### PR TITLE
Open zip archives and empty folders in ColumnView

### LINKED ISSUE #5878: Files won't open empty folders in Column View
### Description

If a folder is completely empty (including any hidden files or folders), Files will not open it's contents in Column View by creating another column, and the File Path Bar will not change from the previously selected folder.

### Steps To Reproduce

1) Create any folder in any application (File Explorer, Files) and leave it empty.
2) Open Files and display the hierarchy the folder is in with Column View.
3) Attempt to view it's contents by double-clicking.

### Expected behavior

The expected behaviour is that **the contents of the folder, even if empty will be displayed in another column in order for you to perform actions inside it.**  However this is not the case.

In the uploaded image you can see that even though the "Files Playground" folder is the currently active one, the File Path Bar has not changed to reflect that and the contents are not being displayed.

Even if something is added to the folder (either using Files or another app), you will need to use the Refresh action to be able to then view it's contents.

### Files Version

Version: 1.9.25.0

### Windows Version

Windows 11 Pro 21H2 22000.160

### Relevant Assets

image

### LINKED ISSUE #5905: Files crashes on Windows 10 1809
### Description

The app takes a long time to populate the side panel / drives.
Trying to open a folder crashes the app.

### Steps To Reproduce

1. Open Files.
2. Try navigating to a folder.
3. Observe the app crashing.

### Expected behavior

The app should be able to navigate into folders.

### Files Version

v1.9.26.0

### Windows Version

Windows 10 1809 17763

### Relevant Assets

[debug.log](https://github.com/files-community/Files/files/7032411/debug.log)

https://user-images.githubusercontent.com/8061077/130458757-d3b412ea-0ac7-4085-b822-104132968096.mp4

### LINKED ISSUE #5928: When a column in Details View is automatically expanded the scroll column becomes hidden
### Description

https://user-images.githubusercontent.com/9108903/130665528-9ab04613-3ecb-4f96-a980-6f2d770946d9.mp4

When the expanded contents of all columns in Details View become larger than the window, the scroll bar is moved off the visible window space until the column widths or window size is interacted with again.

### Steps To Reproduce

1) Make the window small-ish
2) Double-click on a boundary between two columns in Details View

### Expected behavior

The scroll bar should always be visible no matter the circumstances.

### Files Version

Version: 1.9.27.0

### Windows Version

Windows 11 Pro 21H2 22000.160

### Relevant Assets

_No response_

### PR DESCRIPTION
**Pull request by gave92**
**Resolved / Related Issues**
Items resolved / related issues by this PR.
- Closes #5878, #5908
- Closes #5905
- Closes #5928

**Details of Changes**
Add details of changes here.
- Open zip archives and empty folders in ColumnView
- Add API contract check for ShouldConstrainToRootBounds flyout property 
- Removed unused API contract checks in xaml (e.g contract 7 checks not needed since min version == 1809 == contract 7)

**Validation**
How did you test these changes?
- [x] Built and ran the app

### CODE REVIEW

**Review by yaichenbaum — APPROVED** [review]

**Review by winston-de — COMMENTED** [review]
should some indication be shown when a folder is empty?

**Comment by yaichenbaum** [comment]
> should some indication be shown when a folder is empty?
> this snipped should work
> `<uc:FolderEmptyIndicator Grid.Row="3" Margin="0,125,0,0" HorizontalAlignment="Center" VerticalAlignment="Top" Canvas.ZIndex="0" EmptyTextType="{x:Bind ParentShellPageInstance.FilesystemViewModel.EmptyTextType, Mode=OneWay}" />`

That's a good idea but what's the purpose of the margin?

**Comment by winston-de** [comment]
the margin keeps it from being directly at the top of the page

**Comment by yaichenbaum** [comment]
> the margin keeps it from being directly at the top of the page (may not be needed)

Shouldn't it be centered vertically?

**Comment by gave92** [comment]
⁉️

**Comment by yaichenbaum** [comment]
> ⁉️

@gave92 what's the question mark on 😆

**Comment by gave92** [comment]
> should some indication be shown when a folder is empty?
this snipped should work

Lol, confused about this discussion and how it's related to this PR

**Comment by winston-de** [comment]
> > should some indication be shown when a folder is empty?
> this snipped should work
> 
> Lol, confused about this discussion and how it's related to this PR and if it's a request to change something.
> 
> Edit: aaah you mean that no text is shown for an empty folder in column view?

Yeah thats what i meant, sorry for not being clear lol

**Review by winston-de — DISMISSED** [review]

File: Files/Views/LayoutModes/ColumnViewBase.xaml [hunk 1]
```diff
@@ -127,6 +126,13 @@
         </Style>
     </local:BaseLayout.Resources>
     <Grid x:Name="RootGrid" ContextFlyout="{x:Bind BaseContextMenuFlyout}">
+        <uc:FolderEmptyIndicator
+            Margin="0,125,0,0"
```
Comment (yaichenbaum) [inline Files/Views/LayoutModes/ColumnViewBase.xaml:130]:
```suggestion
```

File: Files/Views/LayoutModes/ColumnViewBase.xaml [hunk 2]
```diff
@@ -127,6 +126,13 @@
...
+            HorizontalAlignment="Center"
+            VerticalAlignment="Top"
```
Comment (yaichenbaum) [inline Files/Views/LayoutModes/ColumnViewBase.xaml:132]:
```suggestion
            VerticalAlignment="Center"
```

**Review by yaichenbaum — CHANGES_REQUESTED** [review]

**Review by yaichenbaum — APPROVED** [review]
Great work!

**Pull request merged**